modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/config/DynamicConfigurationProvider.java

```diff
@@ -145,34 +140,23 @@ public ConfigurationInstance get(Event event) {
   private ConfigurationInstance getConfiguration(Pair<ResolverSetResult, ResolverSetResult> resolverSetResult, CoreEvent event)
       throws Exception {
 
-    ConfigurationInstance configuration;
-    cacheReadLock.lock();
-    try {
-      configuration = cache.get(resolverSetResult);
-      if (configuration != null) {
-        // important to account between the boundaries of the lock to prevent race condition
-        updateUsageStatistic(configuration);
-        return configuration;
-      }
-    } finally {
-      cacheReadLock.unlock();
-    }
-
-    cacheWriteLock.lock();
-    try {
-      // re-check in case some other thread beat us to it...
-      configuration = cache.get(resolverSetResult);
-      if (configuration == null) {
-        configuration = createConfiguration(resolverSetResult, event);
-        cache.put(resolverSetResult, configuration);
-      }
-
-      // accounting here for the same reasons as above
+    ConfigurationInstance configuration = cache.getIfPresent(resolverSetResult);
+    if (configuration != null) {
+      // important to account between the boundaries of the lock to prevent race condition
       updateUsageStatistic(configuration);
```

Comment:
these statistics shouldn'e be needed here, they are already handled by the cache, in theory